pkg/codegen/internal/test/testdata/functions.pp.cs

```diff
@@ -0,0 +1,15 @@
+using Pulumi;
+
+class MyStack : Stack
+{
+    public MyStack()
+    {
+        var encoded = Convert.ToBase64String(%v)("haha business");
```

Comment:
This doesn't look right.


1. The `System` namespace needs to be imported for `Convert` or it needs to be fully qualified like `System.Convert`.
2. The `%v` doesn't look right.
3. We can't just pass a string to `Convert.ToBase64String`. The string needs to be converted to `[]byte` or `ReadOnlySpan<byte>`. How are we encoding the string to bytes? UTF-8 given the other languages?

I think we want this to look something like:

```suggestion
        var encoded = Convert.ToBase64String(new System.Text.UTF8Encoding(false).GetBytes("haha business"));
```

The above assumes a `using System;` is added at the top.

Note, unfortunately the `System.Text.Encoding.UTF8` instance emits a byte order mark, so that's why we have to create a specific instance of `UTF8Encoding` and pass `false` for `encoderShouldEmitUTF8Identifier`.

Reply:
Thank you Justin!

Reply:
Actually, I misspoke about the byte order mark. I updated my comment above.

The default `Encoding.UTF8` instance will emit a byte order mark when used with something like `System.IO.StreamWriter`, where the encoding's `GetPreamble()` is called.

If you're just calling `GetBytes` on the encoding, it won't include the byte order mark.

So we can simply do:

```c#
var encoded = Convert.ToBase64String(System.Text.Encoding.UTF8.GetBytes("haha business"));
```

Reply:
And just tried it out to confirm:

```c#
using System;

class Program
{
    static void Main()
    {
        Console.WriteLine(Convert.ToBase64String(new System.Text.UTF8Encoding(false).GetBytes("haha business")));
        Console.WriteLine(Convert.ToBase64String(System.Text.Encoding.UTF8.GetBytes("haha business")));
    }
}
```

Have the same output:

```
$ dotnet run
aGFoYSBidXNpbmVzcw==
aGFoYSBidXNpbmVzcw==
```